protocols/relay/src/priv_client.rs

```diff
@@ -387,10 +386,14 @@ impl NetworkBehaviour for Behaviour {
     }
 }
 
-/// A [`NegotiatedSubstream`] acting as a [`Connection`].
-pub enum Connection {
+pub struct Connection {
+    state: ConnectionState,
+}
+
+/// A [`NegotiatedSubstream`] acting as a [`ConnectionState`].
+pub enum ConnectionState {
```

Comment:
This should be private!

```suggestion
enum ConnectionState {
```

Reply:
@thomaseizinger Done, at first I hadn't because passing private here I got a weird compiler error that says `drop_notifier` is never read elsewhere when it isn't: it's used for example in `new_inbound` and `new_outbound`. I then placed a `#[allow(dead_code)]`